lib/bundles/inspec-compliance/api.rb

```diff
@@ -238,5 +242,11 @@ def self.server_version_from_config(config)
       return nil unless config['version'].is_a?(Hash)
       config['version']['version']
     end
+
+    def self.determine_server_type(url, insecure)
+      return :automate if Compliance::HTTP.get(url + '/compliance/version', nil, insecure).code == '401'
+      return :compliance if Compliance::HTTP.get(url + '/api/version', nil, insecure).code == '200'
+      raise CannotDetermineServerType, "Unable to determine if #{url} is a Chef Automate or Chef Compliance server"
```

Comment:
Just tossing out a thought here... I prefer to have exceptions raised at the point of the implementation where "it matters" (for lack of a better phrase). I'd almost prefer to see this method return `nil` if it can't figure out the server type, and let the caller determine how to handle. That way other methods can use this if needed without having to worry about the stack unwinding on them.

```ruby
def self.determine_server_type(url, insecure)
  if Compliance::HTTP.get(url + '/compliance/version', nil, insecure).code == '401'
    :automate
  elsif Compliance::HTTP.get(url + '/api/version', nil, insecure).code == '200'
    :compliance
  end
end
```